### PR TITLE
Add overview documentation for IPv6

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/area documentation
/kind enhancement
/platform aws

**What this PR does / why we need it**:

Add overview documentation for IPv6.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc user
Add overview documentation for IPv6
```

### CODE REVIEW

File: docs/usage/ipv6.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+## IPv6 Ingress for IPv4 Shoot Clusters
+
+Per default, Gardener shoot clusters use only IPv4.
+Therefore, they also expose their services only via load balancers with IPv4 addresses.
+To allow external clients to also use IPv6 to access services in an IPv4 shoot cluster, the cluster needs to be configured to support dual-stack ingress.
+
+It is possible to configure a shoot cluster to support dual-stack ingress, see [Using IPv4/IPv6 (dual-stack) Ingress in an IPv4 single-stack cluster](dual-stack-ingress.md) for more information.
+
+The main benefit of this approach is that the existing cluster stays almost as is without mayor changes, keeping the operational simplicity.
```
Comment (kon-angelo) [inline docs/usage/ipv6.md:20]: mayor -> `major`

File: docs/usage/ipv6.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      networks:
+        vpc:
+          cidr: 192.168.0.0/16
+        zones:
+          - name: ...
+            public: 192.168.32.0/20
+            internal: 192.168.48.0/20
+```
+
+Please note that `nodes`, `pods` and `services` should not be specified in `.spec.networking` resource.
```
Comment (kon-angelo) [inline docs/usage/ipv6.md:67]: I would put something like

> [!WARNING]  
> Please note that `nodes`, `pods` and `services` should not be specified in `.spec.networking` resource.

File: docs/usage/ipv6.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+Please note that traffic going through NAT64 incurs the same cost as ordinary NAT traffic in an IPv4-only cluster.
+Therefore, it might be beneficial to prefer IPv6 for services, which provide IPv4 and IPv6.
+
+## Dual-Stack Shoot Clusters
+
+### Motivation
+
+Dual-stack shoot clusters support IPv4 and IPv6 out-of-the-box.
+They might also be the obvious intermediate step on the way towards IPv6 for any existing (IPv4-only) clusters.
```
Comment (kon-angelo) [inline docs/usage/ipv6.md:95]: > They might also be the obvious intermediate step

Language like `obvious` can be a bit snarky for a remark.
```
They can be used as an intermediate step
```

File: docs/usage/ipv6.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+It is worth recognizing that the migration from an IPv4-only shoot cluster to a dual-stack shoot cluster involves rolling of the nodes/workload as well.
+Nodes will not get a new IPv6 address assigned automatically.
+The same is true for pods as well.
+Once the migration is supported, the detailed caveats will be documented here.
+
+### Load Balancer Configuration
+
+The AWS Load Balancer Controller is automatically deployed when using a dual-stack shoot cluster.
+When creating a load balancer, the corresponding annotations need to be configured, see [AWS Load Balancer Documentation - Network Load Balancer](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/nlb/) for details.
```
Comment (kon-angelo) [inline docs/usage/ipv6.md:149]: same as above regarding version.

File: docs/usage/ipv6.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+The AWS Load Balancer Controller allows dual-stack ingress so that a dual-stack shoot cluster can serve IPv4 and IPv6 clients.
+You can find an example [here](dual-stack-ingress.md#creating-an-ipv4ipv6-dual-stack-ingress).
+
+Please note that load balancer services without any special annotations will default to IPv4-only regardless how `.spec.ipFamilies` is set.
```
Comment (kon-angelo) [inline docs/usage/ipv6.md:154]: Maybe a warning here too and potentially moving it a bit up at the point where you mention that they need to use the NLB annotations.

File: docs/usage/ipv6.md (hunk 4 above)
Comment (ScheererJ) [reply docs/usage/ipv6.md]: Adapted all links to the AWS load balancer controller documentation.
PTAL.

**Comment by ScheererJ** [comment]
/hold
Will add some details concerning hair-pinning.

File: docs/usage/ipv6.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            public: 192.168.32.0/20
+            internal: 192.168.48.0/20
+```
+
+> [!WARNING]
+> Please note that `nodes`, `pods` and `services` should not be specified in `.spec.networking` resource.
+
+In contrast to that, it is still required to specify IPv4 ranges for the VPC and the public/internal subnets.
+This is mainly due to the fact that public/internal load balancers still require IPv4 addresses as there are no pure IPv6-only load balancers as of now.
+The ranges can be sized according to the expected amount of load balancers per zone/type.
```
Comment (DockToFuture) [inline docs/usage/ipv6.md:72]: Maybe we could mention that the IPv6 ranges are assigned by the hyperscaler via prefix delegation.

File: docs/usage/ipv6.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+### Load Balancer Configuration
+
+The AWS Load Balancer Controller is automatically deployed when using an IPv6-only shoot cluster.
```
Comment (DockToFuture) [inline docs/usage/ipv6.md:76]: The AWS Load Balancer Controller is automatically deployed when using an IPv6 or dual stack shoot cluster.

File: docs/usage/ipv6.md (hunk 6 above)
Comment (ScheererJ) [reply docs/usage/ipv6.md:72]: Addressed with https://github.com/gardener/gardener-extension-provider-aws/pull/1143/commits/1c2149370fe2b6e6fca0402cf3e2ca367ac3c4a5.

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**